src/common/darktable.c

```diff
@@ -770,6 +770,30 @@ int dt_init(int argc, char *argv[], const gboolean init_gui, const gboolean load
     gtk_init(&argc, &argv);
   }
 
+  // execute a performance check and configuration if needed
+  int last_configure_version = dt_conf_get_int("performance_configuration_version_completed");
+  if(last_configure_version < DT_CURRENT_PERFORMANCE_CONFIGURE_VERSION)
+  {
+    // if this is the very first time -> run this configuration unconditionally
+    bool run_configure = TRUE;
+
+    // ask the user whether he/she would like
+    // dt to make changes in the settings
+    char *label_text = g_markup_printf_escaped(
+        _("We have an updated performance configuration logic - executing that might improve the "
```

Comment:
Some typos in that string. Fixed version:

> We have an updated performance configuration logic - executing that might improve the performance of darktable.\nThis will potentially overwrite some of your existing settings - especially in case you have manually modified them to custom values.\nWould you like to execute this update of the performance configuration?\n